### DISCUSSION TITLE
License

### DISCUSSION THREAD

**Discussion by Guilherme-Cassel** [QuestPDF/QuestPDF#846] [General]
I work in a medium-size company, and just me and my colleage decided to develop something on it, so we're 2, following our hobby and turning it in something useful.

Through our journey into making an internal software for the company, to store all automatizations, we found QuestPDF, and its gorgeously useful, but paid.

Since I care so much about any spendings on company, I wanted to know, are we allowed to use at least version 2022.12 without paying professional licence?

We would use it for extracting database information and then print reports of it, for company's internal use, no external distribution or selling any software with it.

Or even, are we allowed to use newer versions with MIT license? Because seems like the use we would give to it is not exactly for profit, just to make everyone's lifes better with the cool reports. Tell me if I'm wrong.

I've read a lot from QuestPDF documentation to understand the licensing, but still wanted confirmation if my actions would be right.

**Comment by MarcinZiabek** [comment]
> I work in a medium-size company, and just me and my colleage decided to develop something on it, so we're 2, following our hobby and turning it in something useful.
> 
> Through our journey into making an internal software for the company, to store all automatizations, we found QuestPDF, and its gorgeously useful, but paid.

Thank you so much for sharing your positive feedback regarding the library. I understand that the hybrid license may not be desirable but it is the only way to keep it alive and sustainable.

> Since I care so much about any spendings on company, I wanted to know, are we allowed to use at least version 2022.12 without paying professional licence?

I am happy to confirm your understanding. The `2022.12.X` versions are released under the MIT license and can be used for free forever. I still plan to release one more minor maintenance update, related mainly to updated dependencies and checking the compatibility. The official support will be dropped in Q2 2024.

> We would use it for extracting database information and then print reports of it, for company's internal use, no external distribution or selling any software with it.
> 
> Or even, are we allowed to use newer versions with MIT license? Because seems like the use we would give to it is not exactly for profit, just to make everyone's lifes better with the cool reports. Tell me if I'm wrong.

Unfortunately, I'm afraid I have to disagree. This is a pretty grey area, isn't it? In my interpretation, every action within the company that leads to internal improvements and process optimizations (such as generating PDF reports) is, in some sense, done for profit. I hope you understand my point of view 😄 

Thank you for asking about license details. I appreciate your desire to stay fully compliant. Have a wonderful day!

**Reply by Guilherme-Cassel** [reply]
Thanks, you're right! I was 95% sure of the answer you'd give. but just wanted confirmation!

Actually I love softwares that have those kind of licencing.. like.. if you're an student or non-profit, you can use it for free. Like visual studio for example, which the community licence doesnt lack any feature compared to paid licence, just like QuestPDF.

Version 2022.12 is not that bad actually It's pretty complete, but of course I miss some features such as merge, etc...

**Comment by JSR33** [comment]
@MarcinZiabek I'm currently working for a company with more than 10 devs and I'd like to confirm (the email contact button was not working for me) that the pro license is perpetual but for example if the company purchase the license today it will only be able to make use of new versions until 12/19/2025. After that date if we want to use an earlier version we need to pay for a new license. Am I understanding it correctly?

**Reply by MarcinZiabek** [reply]
I’m glad to say “no” to subscriptions! 😄

The license is perpetual and valid for a lifetime. Once purchased, you can use it indefinitely and enjoy one year of free feature, quality, and security updates. It’s not a subscription. After that first year, if you’d like to upgrade to the newest version, you can choose to renew the license.

Of course, the idea behind offering paid licenses is to ensure the necessary funding for ongoing development. I will do my best to give you plenty of reasons to renew your license and continue enjoying an excellent coding experience! 😇

**Reply by JSR33** [reply]
And there is any discount for renewals? 😇

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

We hope the replies above were helpful. For any further licensing questions, the full details are available in the [License](https://www.questpdf.com/license/) section of the documentation, and the team can always be reached directly via the contact form on the website.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
